packages/components/src/components/hds/app-footer/status-link.ts

```diff
@@ -58,16 +58,11 @@ export default class HdsAppFooterStatusLinkComponent extends Component<HdsAppFoo
     return status;
   }
 
-  /**
-   * @param statusIcon
-   * @type {string}
-   * @description The name for the StatusLink icon
-   */
-  get statusIcon(): FlightIconSignature['Args']['name'] {
-    if (this.status && !this.args.statusIcon) {
-      return STATUSES[this.status]?.iconName;
-    }
-    return this.args.statusIcon;
+  get statusIcon(): HdsIconSignature['Args']['name'] | undefined {
+    return (
+      this.args.statusIcon ??
+      (this.status !== undefined ? STATUSES[this.status]?.iconName : undefined)
+    );
```

Comment:
thought: I don't think the new logic is equivalent to the old one, can you double check?

Reply:
Sure, here's some test cases I put together with copilot

```
1. Both status and statusIcon are undefined:
Input: this.status = undefined, this.args.statusIcon = undefined
Expected Output: undefined

2. status is defined, statusIcon is undefined:
Input: this.status = 'operational', this.args.statusIcon = undefined
Expected Output: STATUSES['operational']?.iconName

3. status is undefined, statusIcon is defined:
Input: this.status = undefined, this.args.statusIcon = 'custom-icon'
Expected Output: 'custom-icon'

4. status and statusIcon are both defined:
Input: this.status = 'ON_TIME', this.args.statusIcon = 'custom-icon'
Expected Output: 'custom-icon'

5.status is null, statusIcon is undefined:
Input: this.status = null, this.args.statusIcon = undefined
Expected Output: undefined

6. status is a valid string that does not exist in STATUSES, and statusIcon is undefined:
Input: this.status = 'UNKNOWN', this.args.statusIcon = undefined
Expected Output: undefined (assuming STATUSES['UNKNOWN'] does not exist)

7. status is undefined, statusIcon is a string with a space:
Input: this.status = undefined, this.args.statusIcon = ' '
Expected Output: ' ' (assuming it's a valid icon name)
```

Reply:
@KristinLBradley would you be able to help check this change when you have the time (I believe you're the initial author)

Reply:
the old code reads easier to me, but not strongly opinionated; this is how I read it

> if we have a status but no custom icon (`statusIcon`) return the predefined icon for that status.
> otherwise return the custom icon (`statusIcon`)

I'd ask Kristin or someone else to assess the legibility of the new code.

Reply:
Tagged Kristin to take a look.

The way I see it is:

- If we have a `@statusIcon`, we always want to use it.
- If we dont:
  - If we have a status, use it to access available STATUSES, but if not, return nothing

Reply:
@zamoore For me, it's a little harder to read the logic but it's more explicit about what it returns so that's a benefit.

Reply:
@KristinLBradley Thanks for taking a look. That was exactly the intention. It became easier to satisfy typescript after the change. If there are no hard objections, I'm going to leave it as is.

I appreciate the input from both of you!